### PR TITLE
Initial sources commit

### PR DESCRIPTION
**Pull request by pzaino**
This PR contains:
- [ ] bugfix
- [x] feature
- [ ] refactor
- [ ] documentation
- [ ] other

Are tests included?
- [ ] no
- [x] yes (*bugfixes and features will not be merged without tests*)

Breaking Changes?
- [x] no
- [ ] yes (*breaking changes will not be merged unless absolutely necessary – please provide motivation*)

List any relevant issue numbers:

### Description
Guys initial push of ported code here, we still need to work out a good test automation, we can open a discussion about it, for now let's see if this initial commit looks good for you, thanks.

### CODE REVIEW

**Review by InteractiveYelp — COMMENTED** [review]
Build
-----

I have not built it, as MkGCC doesn't work. make starts, does some RMEnsures (for UnixLib), and then sits there using all the processor without any output (gcc is never started). (make 4.2.1 built for arm-unknown-riscos).

Files
-----

What is the empty models directory for?

src./gitkeep is not needed.

!LibEzINI.o contains two already compiled library files. Is this intentional, as they are not mentioned in the ReadMe, and it's not clear what the difference between them is? I would have expected those in a release archive instead -- you don't have any script for cleaning that after running MkGCC before committing; just /gitignore?

Commit message
--------------

Can you provide a detailed commit message. In particular, the library version number, where you got it, and what changes (if any) you made to it.

Doxygen
-------

The files src.docs appear to be auto-generated by Doxygen. These do not form part of the source code, so can you move that somewhere else, say to doc.docs? Also I'm not sure about having 172 auto-generated files in the repository. Maybe put them in an archive, even as a separate release on github?

File: !LibEzINI/!Help [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+- Port by Paolo Fabio Zaino 2020
+---------------------------------------------------------------
+
+_ Release 1.0.0 _
+
+-----------------------
+-  DISCLAIMER  -
+-----------------------
+
+This is a port of a LIbrary under GPL license therefore the port inherit the license and so you are also required to respect the original author license.
```
Comment (skymandr) [inline !LibEzINI/!Help]:
```suggestion
This is a port of a Library under LGPL v3 license, therefore the port inherits the license.
```
I think the extra text is unnecessary.

File: !LibEzINI/!Help [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+If you do not know what an INI fil is you probably do not need this library either.
+
+If you want to know what an INI file is, it's simple. INI files are generally used to store applications configurations (that's why the INI term, which stays for INItiatization)
+
+Their structure is super simple:
+
+<key> = <value>
+
+They pre-date JSON and YAML formats and are still widely used on may platforms. The most famous one was old Windows OS which used the INI files for a lot of things. Linux also used them to allow configuration of services and applications.
```
Comment (skymandr) [inline !LibEzINI/!Help]:
```suggestion
This library is a simple C library to read/write and use INI files on RISC OS and in your applications.

If you do not know what an INI file is you probably do not need this library either.

If you want to know what an INI file is, it's simple. INI files are generally used to store applications configurations (hence the INI term, which stands for INItiatization)

Their structure is super simple:

<key> = <value>

INI files pre-date JSON and YAML formats and are still widely used on may platforms. The most famous one was he old Windows OS which used the INI files for a lot of things. Linux also used them to allow configuration of services and applications.
```

File: !LibEzINI/!Help [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+Here is another example:
+
+hostname = "mycomputer"
+
+So again it's simple.
+
+---------------
+- FINAL NOTES -
+---------------
+Thanks for evaluating my port and effort and eventually use it in your code, and, most of all, thanks for coding on RISC OS :)
```
Comment (skymandr) [inline !LibEzINI/!Help]:
```suggestion
Thanks for evaluating my port and effort and eventually using it in your code, and, most of all, thanks for coding on RISC OS :)
```

File: README.md [hunk 4]
```diff
@@ -1 +1,48 @@
...
+This library is intended to be easy to use and build upon.  It is small in size, thus not as feature complete as other libraries, but should be sufficient to meet most requirements.
+
+## Build the lib
+To build from the source just git clone this repo and then go to LibEzINI -> src directory.
+
+### Using GCC
+To build these files with GNU make and gcc, simply double click on the file called MkGCC, before you do that make sure that your RISC OS device Filer has seen where your copy of !GCC is (usually in Apps.Development).
+
+### Using ROOL DDE
+To build these files with ROOL DDE, simply click on the file called MkDDE, before you do that please make sure your RISC OS device Filer has executed the !SetPaths from DDE which shoul dbe placed in AcornC/C++ directory.
```
Comment (skymandr) [inline README.md]:
```suggestion
To build these files with ROOL DDE, simply click on the file called MkDDE, before you do that please make sure your RISC OS device Filer has executed the !SetPaths from DDE which should be placed in AcornC/C++ directory.
```

File: README.md [hunk 5]
```diff
@@ -1 +1,48 @@
...
+
+Write an INI file by using AddEntryToList to build an entry list, then call
+MakeINIFile to make a file from the entry list.  Call FreeList when you are
+done.
+
+## Feedback
+Feedback is always welcome!
+
+### In case of problems
+For any issue you may encounter please use the Issues option here on GitHub on the top. Please do not try to contact us directly, the RISC OS Community handles all the communications here on github.
```
Comment (skymandr) [inline README.md]:
```suggestion
For any issue you may encounter please use the Issues option here on GitHub. You can find it at the top of the page. Please do not try to contact us directly, the RISC OS Community handles all the communications here on GitHub.
```

File: README.md [hunk 6]
```diff
@@ -1 +1,48 @@
...
+
+### Requesting new features
+As for the problems reporting please use the Issues option here on GitHub also to request new features.
+
+### Contributing
+We welcome improvements and new ideas, before you submit your changes please have a look at the Contributing Guidelines [here](../CONTRIBUTING.md)
+
+## Official ExINI Documentation
```
Comment (skymandr) [inline README.md]:
```suggestion
## Official EzINI Documentation
```

File: LICENSE [hunk 7]
```diff
@@ -0,0 +1,674 @@
+                    GNU GENERAL PUBLIC LICENSE
```
Comment (skymandr) [inline LICENSE]: EzINI is actually LGPL v3, not GPL V3, so this file should be changed. See `src/COPYING.LESSER`.

File: src/README.md [hunk 8]
```diff
@@ -0,0 +1,48 @@
+# LibEzINI
```
Comment (skymandr) [inline src/README.md]: This file is almost but not quite a copy of the README.md in the root. I don't think it is good to keep both, in the interest of maintainability – they have already diverged.

File: src/README.md [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+This is a RISC OS port of EzINI a small adn easy to use ANSI C library implementing INI file reading and writing.
+This library is intended to be easy to use and build upon.  It is
+small in size, thus not as feature complete as other libraries, but should be sufficient to meet most requirements.
```
Comment (skymandr) [inline src/README.md]:
```suggestion
This is a RISC OS port of EzINI a small and easy to use ANSI C library implementing INI file reading and writing.
This library is intended to be easy to use and build upon.  It is
small in size, thus not as feature complete as other libraries, but should be sufficient to meet most requirements.
```

File: src/README.md [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+
+Write an INI file by using AddEntryToList to build an entry list, then call
+MakeINIFile to make a file from the entry list.  Call FreeList when you are
+done.
+
+## Feedback
+Feedback is always welcome!
+
+### In case of problems
+For any issue you may encounter please use the Issues option here on GitHub on the top. Please do not try to contact us directly, the RISC OS Community handles all the communications here on github.
```
Comment (skymandr) [inline src/README.md]: see comments on other README.md above.

File: src/README.md [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+small in size, thus not as feature complete as other libraries, but should be sufficient to meet most requirements.
+
+## Build the lib
+To build from the source just git clone this repo and then go to LibEzINI -> src directory.
+
+### Using GCC
+To build these files with GNU make and gcc, simply double click on the file called MkGCC, before you do that make sure that your RISC OS device Filer has seen where your copy of !GCC is (usually in Apps.Development.
+
+### Using ROOL DDE
+To build these files with ROOL DDE, simply click on the file called MkDDE, before you do that please make sure your RISC OS device Filer has executed the !SetPaths from DDE which shoul dbe placed in AcornC/C++ directory.
```
Comment (skymandr) [inline src/README.md]:
```suggestion
To build these files with ROOL DDE, simply click on the file called MkDDE, before you do that please make sure your RISC OS device Filer has executed the !SetPaths from DDE which should be placed in AcornC/C++ directory.
```

File: src/README.md [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+As for the problems reporting please use the Issues option here on GitHub also to request new features.
+
+### Contributing
+We welcome improvements and new ideas, before you submit your changes please have a look at the Contributing Guidelines [here](CONTRIBUTING.md)
+
+## Official ExINI Documentation
+See https://michaeldipperstein.github.io/ezini/
+
+## License
+The original EzINI is released under GPLv3, so is LibEzINI port here.
```
Comment (skymandr) [inline src/README.md]:
```suggestion
## Official EzINI Documentation
See https://michaeldipperstein.github.io/ezini/

## License
The original EzINI is released under GPLv3, so is this LibEzINI port.
```

File: src/README.md (hunk 8 above)
Comment (skymandr) [reply src/README.md]: I am repeating some but not all comments on that file here, so please make sure to take all comments into account, whether you keep one or both files.

**Review by skymandr — CHANGES_REQUESTED** [review]

File: src/README.md (hunk 8 above)
Comment (pzaino) [reply src/README.md]: sure, and good point!

File: src/README.md (hunk 10 above)
Comment (pzaino) [reply src/README.md]: k, will remove the readme in the src

File: src/README.md (hunk 8 above)
Comment (pzaino) [reply src/README.md]: ok removed the file, so I guess this is resolved

**Comment by skymandr** [comment]
Also I agree with @InteractiveYelp  that it is questionable whether the doc-folder is worth keeping. Any user should be able to build it from source, right? If we can build it, it can be packaged in with pre-built releases instead.

**Comment by pzaino** [comment]
@InteractiveYelp ,
Thanks for your comments, for some reasons I don't seems to be able to reply them directly.

Anyway:

> I have not built it, as MkGCC doesn't work. make starts, does some RMEnsures (for UnixLib), and then sits there using all the processor without any output (gcc is never started). (make 4.2.1 built for arm-unknown-riscos).

I have the same make release, it seems a bug in the GNU Make utility that was triggered by using Prefix \<Obey$Dir\> instead of Dir. Replaced with Dir now and testing on various boards. To me it was working, so I guess it's something influenced by the presence of ROOL DDE on the same machine?

> What is the empty models directory for?

Usually to standardise the repository directory tree (all repos presents same directory structure, makes it easy for people to find things), if we don't want to follow this, I am ok, let put it to vote across us and follow what the majority wants, I am easy :)

> src./gitkeep is not needed.

.gitkeep it's kind of a GIT standard to pull in empty directories, so its use depends on the decision above

> Can you provide a detailed commit message. In particular, the library version number, where you got it, and what changes (if any) you made to it.

Sure, I thought as an initial commit the readme pointed to enough of these resources, except I think I forgot to put in the readme the EzINI release from where I created the RISC OS spin-off. For the record here it's EzINI 0.4

> The files src.docs appear to be auto-generated by Doxygen. These do not form part of the source code, so can you move that somewhere else, say to doc.docs?

To move them sure, while about they do not form part of the source it's kinda "ish" doxygen generates them using the source code. So we can simply ask Doxygen to generate them in the !LibEzINI or just move them in there after a build.

BTW about Doxygen, we still do not have a release for RISC OS, so for now I am using it from Linux, not optimal I know, but if someone would like to have a look at porting it that would be cool :)

P.S. sorry for all the typos, I just have so much limited time and I am trying to run as fast as I can right now...

**Comment by pzaino** [comment]
> Also I agree with @InteractiveYelp that it is questionable whether the doc-folder is worth keeping. Any user should be able to build it from source, right? If we can build it, it can be packaged in with pre-built releases instead.

So I am ok with the choice guys, but right now RISC OS does not have a port of Doxygen, while I think it's available on GCCSDK (it may just be from Ubuntu, so I need to double check). So yes a user can rebuild the docs from the sources (except for the images I believe, need to double check this too), but only if they have Linux and they make doxygen understand RISC OS source tree... We could help Doxygen by adding the /c at the end of each file name.... but some people think it's ugly...

So I am open to suggestions :)

**Comment by pzaino** [comment]
@InteractiveYelp please let me know if it builds now on your system. I tested on quite few boards here and it builds everywhere

@RISC-OS-Community/code-reviews-team I think I have done all the changes requested, please let me know if I have missed anything, thanks!

**Review by skymandr — COMMENTED** [review]
I'm happy when @InteractiveYelp is happy.

File: LICENSE (hunk 7 above)
Comment (pzaino) [reply LICENSE]: Just saw this and hummm the original author provided both licenses... confusing, but ok let me fix that!

**Comment by pzaino** [comment]
A bit more details on this one:

> What is the empty models directory for?

The models directory usually contains modelling code that we may want to source control. Modelling code is useful when we want to solve a problem from different angles (or different approaches) to find the best approach. Not all modelling code will make it in src. Src contains the actual decided approach. BTW @InteractiveYelp we have an open discussion on the Repository Template, can you join us there please? https://github.com/RISC-OS-Community/StandardRepoTemplate/issues/1

File: LICENSE (hunk 7 above)
Comment (skymandr) [reply LICENSE]: That is probably because LGPL references GPL. As of v3 LGPL is an extension of GPL, I think.

**Comment by pzaino** [comment]
Ok haven't heard any objection in few days, so I am going to merge this pull request, thanks everyone for the feedback!

**Pull request merged**